### ISSUE TITLE
fixed size of icons within buttons

### ISSUE THREAD

**Issue by m-siekaniec** [sl-design-system/components#1594]
### Provide a general summary of the issue here

When rescaling text (to mock behavior of text resize behavior as in wcag 1.4.4) the icons doesn't scale (is locked in the fixed container) which makes the icon-only button keep its size, while the one with text grows. In order to rescale it, I need to detach the component each time. 

### 🤔 Expected Behavior?

When changing an icon size the container should resize as well (probably it should be hugging). 

### 😯 Current Behavior

icon lies in the fixed container

### 💁 Possible Solution

set the icon container to hug

### 📷 Screenshots

![image](https://github.com/user-attachments/assets/7ca193c7-de39-47ed-b1a9-10f1a99ae7c8)

### 👤 Your name

Marcin SIekaniec

### 🎨 Your Theme(s)

TEAS

### Tasks
- [ ] https://github.com/sl-design-system/components/issues/1932
- [x] @roald please contact Marcin to discuss the workflow in Figma

**Comment by DanielleRameau** [comment]
@marcin-siekaniec right now our Figma components are not made for zooming, may I ask why you want to mock this within Figma?

**Comment by marcin-siekaniec** [comment]
I'm trying to mock how different screens/parts will look like on 200% text resize, 12% character spacing, 150% lineheight due to wcag requirements. Here's an example. 
![image](https://github.com/user-attachments/assets/764a6ba1-7502-4d48-8b8c-95114457e10f)
![image](https://github.com/user-attachments/assets/20ed30db-2c8a-4377-8f2a-6ddfab15386b)
 or
![image](https://github.com/user-attachments/assets/92712dad-c992-4ff9-8f19-5febe05cf598)

**Comment by marcin-siekaniec** [comment]
but yeah, I believe it might work for fa-icons, but not sure if svg's could scale. Still we would need to resize the box manually, so maybe there's no good way to mock it.

**Comment by DanielleRameau** [comment]
Hi @marcin-siekaniec , May I ask why you're doing this within Figma? If development implements it in the right way, it should work automatically. Good to mention: we are currently not scaling icons when you're using text-only zoom (Firefox). The regular page zoom will scale the SVG icons.

**Comment by marcin-siekaniec** [comment]
To see how the content will scale and support my design decisions. For example this is how I decided to drop the idea of having pinned header below a certain breakpoint. 
I would also love see icons scaling in the buttons, especially in icon-only, as they can convey some important info.

**Comment by DanielleRameau** [comment]
@m-siekaniec Roald will contact you

**Comment by DanielleRameau** [comment]
Discussed with Avaava: scale icons in combination with text zoom

**Comment by DanielleRameau** [comment]
opened new story #1932 to solve it in code
